File: dotnet/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/AdvancedRemote.cs
namespace BridgeDesignPattern.Example1;

public class AdvancedRemote : Remote
{
	public AdvancedRemote(IDevice device) : base(device)
	{
	}
	public void Mute()
	{
		if (_device.IsEnabled)
			_device.Volume = 0;
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/BridgeDesignPattern/BridgeDesignPattern.Example1/src/AdvancedRemote.ts
import { IDevice } from './IDevice';
import { Remote } from './Remote';

export class AdvancedRemote extends Remote {
  constructor(device: IDevice) { super(device); }

  mute(): void {
    if (this.device.isEnabled) this.device.volume = 0;
  }
}
